In [3]:
!pip install sweetviz

In [4]:
import pandas as pd

In [5]:
import sweetviz
df = pd.read_csv("COVID clinical trials.csv")


In [6]:
df.head()

Rank   NCT Number                                              Title  \
0     1  NCT04785898  Diagnostic Performance of the ID Now™ COVID-19...   
1     2  NCT04595136  Study to Evaluate the Efficacy of COVID19-0001...   
2     3  NCT04395482  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   
3     4  NCT04416061  The Role of a Private Hospital in Hong Kong Am...   
4     5  NCT04395924         Maternal-foetal Transmission of SARS-Cov-2   

        Acronym                  Status         Study Results  \
0   COVID-IDNow  Active, not recruiting  No Results Available   
1      COVID-19      Not yet recruiting  No Results Available   
2   TAC-COVID19              Recruiting  No Results Available   
3      COVID-19  Active, not recruiting  No Results Available   
4  TMF-COVID-19              Recruiting  No Results Available   

                                          Conditions  \
0                                            Covid19   
1                               SARS-CoV-2 Infection   
2                                            covid19   
3                                              COVID   
4  Maternal Fetal Infection Transmission|COVID-19...   

                                       Interventions  \
0   Diagnostic Test: ID Now™ COVID-19 Screening Test   
1    Drug: Drug COVID19-0001-USR|Drug: normal saline   
2  Other: Lung CT scan analysis in COVID-19 patients   
3          Diagnostic Test: COVID 19 Diagnostic Test   
4  Diagnostic Test: Diagnosis of SARS-Cov2 by RT-...   

                                    Outcome Measures  \
0  Evaluate the diagnostic performance of the ID ...   
1  Change on viral load results from baseline aft...   
2  A qualitative analysis of parenchymal lung dam...   
3  Proportion of asymptomatic subjects|Proportion...   
4  COVID-19 by positive PCR in cord blood and / o...   

                               Sponsor/Collaborators  ...         Other IDs  \
0              Groupe Hospitalier Paris Saint Joseph  ...       COVID-IDNow   
1                         United Medical Specialties  ...  COVID19-0001-USR   
2                       University of Milano Bicocca  ...       TAC-COVID19   
3                    Hong Kong Sanatorium & Hospital  ...        RC-2020-08   
4  Centre Hospitalier Régional d'Orléans|Centre d...  ...      CHRO-2020-10   

         Start Date Primary Completion Date   Completion Date  \
0  November 9, 2020       December 22, 2020    April 30, 2021   
1  November 2, 2020       December 15, 2020  January 29, 2021   
2       May 7, 2020           June 15, 2021     June 15, 2021   
3      May 25, 2020           July 31, 2020   August 31, 2020   
4       May 5, 2020                May 2021          May 2021   

       First Posted Results First Posted Last Update Posted  \
0     March 8, 2021                  NaN      March 8, 2021   
1  October 20, 2020                  NaN   October 20, 2020   
2      May 20, 2020                  NaN   November 9, 2020   
3      June 4, 2020                  NaN       June 4, 2020   
4      May 20, 2020                  NaN       June 4, 2020   

                                           Locations Study Documents  \
0  Groupe Hospitalier Paris Saint-Joseph, Paris, ...             NaN   
1       Cimedical, Barranquilla, Atlantico, Colombia             NaN   
2  Ospedale Papa Giovanni XXIII, Bergamo, Italy|P...             NaN   
3  Hong Kong Sanatorium & Hospital, Hong Kong, Ho...             NaN   
4                       CHR Orléans, Orléans, France             NaN   

                                           URL  
0  https://ClinicalTrials.gov/show/NCT04785898  
1  https://ClinicalTrials.gov/show/NCT04595136  
2  https://ClinicalTrials.gov/show/NCT04395482  
3  https://ClinicalTrials.gov/show/NCT04416061  
4  https://ClinicalTrials.gov/show/NCT04395924  

[5 rows x 27 columns]

In [7]:
df.info

<bound method DataFrame.info of       Rank   NCT Number                                              Title  \
0        1  NCT04785898  Diagnostic Performance of the ID Now™ COVID-19...   
1        2  NCT04595136  Study to Evaluate the Efficacy of COVID19-0001...   
2        3  NCT04395482  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   
3        4  NCT04416061  The Role of a Private Hospital in Hong Kong Am...   
4        5  NCT04395924         Maternal-foetal Transmission of SARS-Cov-2   
...    ...          ...                                                ...   
5778  5779  NCT04011644  Mobile Health for Alcohol Use Disorders in Cli...   
5779  5780  NCT04681339  Antibiotic Prescription in Children Hospitaliz...   
5780  5781  NCT04740229  Moderate-intensity Flow-based Yoga Effects on ...   
5781  5782  NCT04804917           3-year Follow-up of the Mind My Mind RCT   
5782  5783  NCT04680000  Chronic Pain Management In Primary Care Using ...   

           Acronym             

In [8]:
df.shape

(5783, 27)

In [9]:
from datetime import datetime

In [32]:
st=pd.Series(df['Start Date'])
res=pd.to_datetime(st)
df['Start']=res.dt.strftime("%Y-%m-%d")

st1=pd.Series(df['Completion Date'])
res1=pd.to_datetime(st1)
df['completion']=res1.dt.strftime("%Y-%m-%d")


df.head()

Rank   NCT Number  \
Start                           
2020-11-09     1  NCT04785898   
2020-11-02     2  NCT04595136   
2020-05-07     3  NCT04395482   
2020-05-25     4  NCT04416061   
2020-05-05     5  NCT04395924   

                                                        Title       Acronym  \
Start                                                                         
2020-11-09  Diagnostic Performance of the ID Now™ COVID-19...   COVID-IDNow   
2020-11-02  Study to Evaluate the Efficacy of COVID19-0001...      COVID-19   
2020-05-07  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   TAC-COVID19   
2020-05-25  The Role of a Private Hospital in Hong Kong Am...      COVID-19   
2020-05-05         Maternal-foetal Transmission of SARS-Cov-2  TMF-COVID-19   

                            Status         Study Results  \
Start                                                      
2020-11-09  Active, not recruiting  No Results Available   
2020-11-02      Not yet recruiting  No Results Available   
2020-05-07              Recruiting  No Results Available   
2020-05-25  Active, not recruiting  No Results Available   
2020-05-05              Recruiting  No Results Available   

                                                   Conditions  \
Start                                                           
2020-11-09                                            Covid19   
2020-11-02                               SARS-CoV-2 Infection   
2020-05-07                                            covid19   
2020-05-25                                              COVID   
2020-05-05  Maternal Fetal Infection Transmission|COVID-19...   

                                                Interventions  \
Start                                                           
2020-11-09   Diagnostic Test: ID Now™ COVID-19 Screening Test   
2020-11-02    Drug: Drug COVID19-0001-USR|Drug: normal saline   
2020-05-07  Other: Lung CT scan analysis in COVID-19 patients   
2020-05-25          Diagnostic Test: COVID 19 Diagnostic Test   
2020-05-05  Diagnostic Test: Diagnosis of SARS-Cov2 by RT-...   

                                             Outcome Measures  \
Start                                                           
2020-11-09  Evaluate the diagnostic performance of the ID ...   
2020-11-02  Change on viral load results from baseline aft...   
2020-05-07  A qualitative analysis of parenchymal lung dam...   
2020-05-25  Proportion of asymptomatic subjects|Proportion...   
2020-05-05  COVID-19 by positive PCR in cord blood and / o...   

                                        Sponsor/Collaborators  ...  \
Start                                                          ...   
2020-11-09              Groupe Hospitalier Paris Saint Joseph  ...   
2020-11-02                         United Medical Specialties  ...   
2020-05-07                       University of Milano Bicocca  ...   
2020-05-25                    Hong Kong Sanatorium & Hospital  ...   
2020-05-05  Centre Hospitalier Régional d'Orléans|Centre d...  ...   

           Primary Completion Date   Completion Date      First Posted  \
Start                                                                    
2020-11-09       December 22, 2020    April 30, 2021     March 8, 2021   
2020-11-02       December 15, 2020  January 29, 2021  October 20, 2020   
2020-05-07           June 15, 2021     June 15, 2021      May 20, 2020   
2020-05-25           July 31, 2020   August 31, 2020      June 4, 2020   
2020-05-05                May 2021          May 2021      May 20, 2020   

            Results First Posted Last Update Posted  \
Start                                                 
2020-11-09                   NaN      March 8, 2021   
2020-11-02                   NaN   October 20, 2020   
2020-05-07                   NaN   November 9, 2020   
2020-05-25                   NaN       June 4, 2020   
2020-05-05                   NaN       June 4, 2020   

                                         

In [43]:
df['Start Month']= pd.DatetimeIndex(df['Start']).month
df['Start Year']= pd.DatetimeIndex(df['Start']).year

df['Completion Month']= pd.DatetimeIndex(df['completion']).month
df['Completion  Year']= pd.DatetimeIndex(df['completion']).year

In [44]:
my_report = sweetviz.analyze([df, "Covid 19"],target_feat='Rank')


C:\Users\Nitij\anaconda3\lib\site-packages\sweetviz\dataframe_report.py:74: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  all_source_names = [cur_name for cur_name, cur_series in source_df.iteritems()]
C:\Users\Nitij\anaconda3\lib\site-packages\sweetviz\dataframe_report.py:109: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  filtered_series_names_in_source = [cur_name for cur_name, cur_series in source_df.iteritems()


                                             |                                             | [  0%]   00:00 ->…

C:\Users\Nitij\anaconda3\lib\site-packages\sweetviz\series_analyzer_numeric.py:25: FutureWarning: The 'mad' method is deprecated and will be removed in a future version. To compute the same result, you may do `(df - df.mean()).abs().mean()`.
  stats["mad"] = series.mad()
C:\Users\Nitij\anaconda3\lib\site-packages\sweetviz\series_analyzer_text.py:19: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for item in to_process.source_counts["value_counts_without_nan"].iteritems():
C:\Users\Nitij\anaconda3\lib\site-packages\sweetviz\series_analyzer_text.py:19: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for item in to_process.source_counts["value_counts_without_nan"].iteritems():
C:\Users\Nitij\anaconda3\lib\site-packages\sweetviz\series_analyzer_text.py:19: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for item in to_process.sou

In [45]:
my_report.show_html('Report_covid19_critical_trial_eda.html')

Report Report_covid19_critical_trial_eda.html was generated! NOTEBOOK/COLAB USERS: the web browser MAY not pop up, regardless, the report IS saved in your notebook/colab files.


# Observtions

# Status

1) 18% trials is completed
2) 49% trials is still recruiting
3) 9% trials are active but not recruiting





# Results

Out of 5783 studies only 36 have attained some results that is less than <<1%



# Conditions

 Major research had been done to find a cure for diseases including COVID -19, Coronavirus, SARS Cov 2


# AGE



Most Studies involve (Adult, Older Adult) Population

Only Child studies are very few

# Phases 


Most relevant takes are that a high number 685 of studies are in the 2nd Phase

450 Studies are in Phase 3

234 Studies are in Phase 4 (Very close to reaching a solution)

Majority Studies have Not Applicable Phase


# Funded Bys


Max Funding is by Industry

NIH - National Institute of Health and US FED has also funded many studies

# Study Type

Most Studies are observational(42%) and interventional(57%)

Rarely do studies follow Expanded Access Methods

# START DATE

Maximum trials are started in 2020 and mainly in March, April,June




# COMPLETION DATE

Half number of studies are completed in 2021, and around 30% studies are completed in 2020 only in the month of December


